akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/FilteredPayload.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.grpc.internal
+
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.persistence.query.typed.EventEnvelope
+import akka.projection.BySlicesSourceProvider
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.scaladsl.Source
+
+import scala.concurrent.Future
+
+/**
+ * INTERNAL API
+ *
+ * Placeholder object for filtered events when producer event push is used
+ */
+@InternalApi
+private[akka] case object FilteredPayload
```

Comment:
We probably need to store something that don't bring in a dependency to akka-projection-grpc. When running the actual consumer with eventsBySlices it would be nice if we can use the ordinary eventsBySlices without having additional transformation of this FilteredPayload back to EventEnvelope.filtered = true. In other words eventsBySlices implementation in akka-persistence-r2dbc should be aware of this and recreate envelope with filtered=true.

I'm not sure what is the best way, but zero bytes seems like a good thing to store. Then maybe a special serializerId   that becomes a special case when creating the EventEnvelope (without deserializing the payload)?

Reply:
Allright, it did feel a bit crufty trying to filter it here. Porting it up to a more central projections-place sounds good.

You mean a special serializer id back in core Akka serialization infra? What would be the in heap representation, like a `akka.persistence.NoEvent`? It would show up if running queries manually against the journal as well.

Reply:
ok, you are right that it's probably better to have a "real" event representation, but let's define that in `akka.persistence`.

Reply:
and we would filter it in akka-persistence-r2dbc queries when creating envelopes, setting that `filtered=true`.

Reply:
Now merged in Akka-persistence-r2dbc but will need another milestone release.

Reply:
Released and bumped in [73a426b](https://github.com/akka/akka-projection/pull/932/commits/73a426bee85383c3012e5f4dc87fc92da4e7324e)